docs/custom_conf.py

```diff
@@ -224,3 +224,10 @@
 .. role:: center
    :class: align-center
 '''
+
+# Define a selector that only adds copy buttons to code blocks without the class `no-copy`
+copybutton_selector = "div:not(.no-copy) > div.highlight > pre"
+
+# Define prompts to be excluded from copying when a copy button is used 
+copybutton_prompt_text = r"^.*?[\$>#]\s+"
```

Comment:
Interesting edge case -- well spotted. How common are directory names like this? I wish people wouldn't name directories this way in the first place but we can't win that battle! I didn't encounter these cases in our docs afaik so didn't consider this. I tested your greedy alternative on a set of examples and it seems to work as before but solves this particular issue. I would be happy to go with your suggestion unless you have identified other pitfalls.

Ideally the person writing the documentation should not have to think about the regex in the background; however, I think it is good practice to test how your copyable material is actually copied/pasted. So perhaps in the docs contribution guidelines there could be a brief explanation as to why we use prompts in our docs (currently), how we use regex to solve the copy/paste issue and what to do if an edge-case is encountered.

I still need to explore in the future if there's a better way to communicate PS versus bash without heavy use of prompts. I don't think including the path in the prompt is always necessary but that's something for another day.

Reply:
> How common are directory names like this? 

I doubt it's a common pattern for directory names :)

Reply:
If you remove the non-greedy marker then in `~$ echo "Hello" > file.txt`, for instance (redirections are a common operation), the prompt would be `~$ echo "Hello" > ` instead of just `~$`. It would be obviously wrong.

Reply:
Very good point.

Reply:
I need to find time for more tests tomorrow but the lazy option worked for the types of prompts and commands that exist in the docs. When I tested the greedy alternative I should have considered a broader set of cases.

Current thinking is: lazy option with some wording in contributing.md about conventions for prompts and testing expected copy behavior.